### ISSUE TITLE
[Bug] Walk on Water letting character walk high in the air

### ISSUE THREAD

**Issue by Numi43** [gta-chaos-mod/ChaosModV#437]
Managed to walk high in the air with Walk on Water active.

Previous effects were set player into random vehicle and launch player up during the Jay Norris death cutscene on TV.

**Comment by TheRealAyCe** [comment]
Do you have a video or screenshot? AFAIK Walk on Water spawns an invisible plane to keep the player above the water level. It may be that you were in an area with an unnatural high water level without there actually being any water. But I'm just guessing.

**Comment by SeanMirrsen** [comment]
Yes there's video, it was posted on Discord and pongo requested the issue be posted here for tracking. It was like the invisible plane kept spawning under the car the player was in, and was launching it upwards, and when the player exited it the plane stayed and the player could walk in the sky. It definitely had to do with the cutscene, something about being set into the nearby vehicle then being launched up, caused the player to detect as being underwater. Maybe the car fell through the world, then while it was down there (and still in cutscene) the walk on water activated, and because it was a cutscene and there was no water anywhere, it didn't quite reset when the cutscene ended.
